### DISCUSSION TITLE
Problem with Prisma in docker production (Segmentation fault)

### DISCUSSION THREAD

**Discussion by FelixBaum** [prisma/prisma#15569] [Q&A] [answered]
Hey everyone,
I have a existing project in which I want to integrate Prisma. On my development machine I initialized Prisma and generated the `schema.prisma` file. In my `Dockerfile` I have two phases, one for building and one for running the server:

```
FROM node:18.9.0-alpine3.15

WORKDIR /api

COPY package.json ./
COPY .npmrc ./
COPY tsconfig.json ./
COPY config.json ./
COPY src ./src
COPY prisma ./prisma

RUN ls -a
RUN npm install
RUN npx prisma generate
RUN npm run build


FROM node:18.9.0-alpine3.15

WORKDIR /api

COPY package.json ./
COPY .npmrc ./
COPY config.json ./
COPY prisma ./prisma

RUN npm install --omit=dev
RUN npx prisma generate

COPY --from=0 /api/dist .

ENV NODE_ENV production
EXPOSE 3000
CMD ["node", "server.js"]
```

**First question:** Do I have to run `npx prisma generate` on both steps?

**Second problem:** The container can successfully start, but if I request a route which uses the generated Prisma client, the process terminates without any errors. I can't figure out, what the problem is. On my development machine, running the server outside of docker, it is working right.

**Update:** Figured out, that the process is terminated with a segmentation fault. Switched the engineType to `binary` and now it's working!

**Comment by nurul3101** [comment]
> Figured out, that the process is terminated with a segmentation fault. Switched the engineType to binary and now it's working!


Glad to see that this is resolved. :+1:

**Comment by janpio** [comment, marked as answer]
That should not be happening. The `binary` engineType should not be something you have to use, so please open a bug report about this @FelixBaum

**Reply by FelixBaum** [reply]
Created an issue here: https://github.com/prisma/prisma/issues/15599 @janpio

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
